src/useWindowScroll.ts

```diff
@@ -9,19 +9,30 @@ export interface State {
 }
 
 const useWindowScroll = (): State => {
-  const [state, setState] = useRafState<State>({
+  const [state, setState] = useRafState<State>(() => ({
     x: isClient ? window.pageXOffset : 0,
     y: isClient ? window.pageYOffset : 0,
-  });
+  }));
 
   useEffect(() => {
     const handler = () => {
-      setState({
-        x: window.pageXOffset,
-        y: window.pageYOffset,
+      setState((state) => {
+        const { pageXOffset, pageYOffset } = window;
+        //Check state for change, return same state if no change happened to prevent rerender
+        //(see useState/setState documentation). useState/setState is used internally in useRafState/setState.
```

Comment:
If referencing the docs, a link inside the comment would be nice :-)
https://reactjs.org/docs/hooks-reference.html#functional-updates